src/DataDepsGenerators.jl

```diff
@@ -19,9 +19,9 @@ struct Metadata
     author::Opt{String}
     maintainer::Opt{String}
     license::Opt{String}
-    publishedDate::Opt{String}
-    createDate::Opt{String}
-    modifiedDate::Opt{String}
+    publishedDate::Union{DateTime, Opt{String}}
+    createDate::Union{DateTime, Opt{String}}
+    modifiedDate::Union{DateTime, Opt{String}}
```

Comment:
It does the same thing, but I think more semantic is
`Opt{Union{DateTime,String}}`